### ISSUE TITLE
no respect to value transformer defined on the fields

### ISSUE THREAD

**Issue by muradm** [typeorm/typeorm#1990]
**Issue type:**

[ ] question
[x ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```
// define transformer column options with transformer
export const UuidValueTransformer = {to: v => uuidParser.parse(v, new Buffer(16)), from: v => uuidParser.unparse(v)};

export const UuidColumnOptions: ColumnOptions = {type: 'binary', length: 16, transformer: UuidValueTransformer};

@Entity()
export class Name {
  @PrimaryColumn('uuid', UuidColumnOptions)
  id: string;

  @Column('varchar')
  name: string;

  @OneToMany(type => Category, role => Category.name)
  categories: Category[];

}

@Entity()
export class Category {

  @Column('varchar')
  category: string;

  @ManyToOne(type => Name, name => name.categories, {primary: true})
  name: Name;

  @Column('uuid', UuidColumnOptions)
  nameId: string;
}

// now try to get relations
const name = await nameRepo.createQueryBuilder().findOne(...);

name.categories = await categoryRepo.createQueryBuilder()
  .relation(Name, 'categories').of(name).loadMany();
```

Result:
```
query: SELECT `categories`.`category` AS `categories_category`, `categories`.`nameId` AS `categories_nameId` FROM `categories` `categories` WHERE `categories`.`nameId` IN (?) -- PARAMETERS: [["28802132-6858-4fd5-bf98-70e0064e340d"]]
```
Query builder don't respect the value transformer defined on the fields.

**Comment by pleerock** [comment]
Correct, value transformers now only work in find methods and query builder. Feel free to contribute!

**Comment by pleerock** [comment]
value transformers is the feature implemented by community, typeorm could live without this feature

**Comment by makepost** [comment]
> value transformers now only work in find methods and query builder

@pleerock, OP uses query builder in his example, so value transformers now only seem to work in find methods, not query builder. Could you please clarify where they can and can't work? Might it be a regression in 0.2?

**Comment by pleerock** [comment]
There can be hundred of cases where value transformers can be applied under different circumstances, so I cant talk about all of them. If you got something not working, please report and try to contribute if you can.

In this particular case they are not working in **relational query builder**, not in select query builder.

**Comment by SylvainEstevez** [comment]
Looks like the PR got closed without being merged @svvac :(

@pleerock If `getEntityValue()` is not the right place to apply transformations (cf https://github.com/typeorm/typeorm/pull/941#issuecomment-332244903), then where do you think  it should happen?

I'll try to provide a PR for this if we can agree on a solution.
